lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -1750,6 +1750,34 @@ void InterpreterFunction::fwdBatchedReduceAddInst(
   }
 }
 
+void InterpreterFunction::fwdLengthsSumInst(const LengthsSumInst *I) {
+  auto out = getTensor(I->getDest());
+  auto data = getTensor(I->getData());
+  auto lengths = getTensor(I->getLengths());
+
+  out->zero();
+
+  auto LH = lengths->getHandle<int64_t>();
+
+  size_t segments = lengths->dims()[0];
+  size_t sliceSize = data->size() / data->dims()[0];
+
+  auto DH = data->getHandle<float>();
+  auto OH = out->getHandle<float>();
+
+  size_t offsetIn = 0;
+  size_t offsetOut = 0;
+  for (size_t i = 0; i < segments; i++) {
+    for (size_t j = 0, e = LH.raw(i); j < e; j++) {
+      for (size_t k = 0; k < sliceSize; k++) {
+        OH.raw(offsetOut + k) += DH.raw(offsetIn + k);
+      }
+      offsetIn += sliceSize;
+    }
+    offsetOut += sliceSize;
+  }
+}
```

Comment:
please assert at the end that resulting offset is equal to corresponding tensor's size. That is, we used the whole tensor.